Xamarin.Forms.Maps.Android/MapRenderer.cs

```diff
@@ -779,6 +789,112 @@ void RemovePolygon(Polygon polygon)
 		}
 
 		#endregion
+
+		#region Circles
+
+		protected virtual CircleOptions CreateCircleOptions(Circle circle)
+		{
+			var opts = new CircleOptions()
+				.InvokeCenter(new LatLng(circle.Center.Latitude, circle.Center.Longitude))
+				.InvokeRadius(circle.Radius.Meters)
+				.InvokeStrokeWidth(circle.StrokeWidth);
+
+			if (!circle.StrokeColor.IsDefault)
+				opts.InvokeStrokeColor(circle.StrokeColor.ToAndroid());
+
+			if (!circle.FillColor.IsDefault)
+				opts.InvokeFillColor(circle.FillColor.ToAndroid());
+
+			return opts;
+		}
+
+		protected ACircle GetNativeCircle(Circle circle)
+		{
+			return _circles?.Find(c => c.Id == (string)circle.MapElementId);
```

Comment:
`.Find()` is not performant on Android. Can you find a way to write this with a for loop instead? Thanks!

Reply:
It's funny because you told someone else to use `.Find()` in this file a year ago. https://github.com/xamarin/Xamarin.Forms/pull/3751#discussion_r216755568

I'll switch everything to loops.

Reply:
That is awesome. Yes, I will slowly but surely move all code to `for` loops. 😂